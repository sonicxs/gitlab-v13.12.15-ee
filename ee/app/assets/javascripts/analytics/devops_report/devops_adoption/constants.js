import { s__, __, sprintf } from '~/locale';

export const DEFAULT_POLLING_INTERVAL = 30000;

export const MAX_SEGMENTS = 30;

export const MAX_REQUEST_COUNT = 10;

export const DEVOPS_ADOPTION_SEGMENT_MODAL_ID = 'devopsSegmentModal';

export const DEVOPS_ADOPTION_SEGMENT_DELETE_MODAL_ID = 'devopsSegmentDeleteModal';

export const DATE_TIME_FORMAT = 'yyyy-mm-dd HH:MM';

export const DEVOPS_ADOPTION_ERROR_KEYS = {
  groups: 'groupsError',
  segments: 'segmentsError',
  addSegment: 'addSegmentsError',
};

export const TABLE_HEADER_TEXT = s__(
  'DevopsAdoption|Feature adoption is based on usage in the current calendar month. Last updated: %{timestamp}.',
);

export const ADD_REMOVE_BUTTON_TOOLTIP = sprintf(
  s__('DevopsAdoption|Maximum %{maxSegments} groups allowed'),
  {
    maxSegments: MAX_SEGMENTS,
  },
);

export const DEVOPS_ADOPTION_GROUP_LEVEL_LABEL = s__('DevopsAdoption|Add/remove sub-groups');

export const DEVOPS_ADOPTION_TABLE_REMOVE_BUTTON_DISABLED = s__(
  'DevopsAdoption|You cannot remove the group you are currently in.',
);

export const DEVOPS_ADOPTION_STRINGS = {
  app: {
    [DEVOPS_ADOPTION_ERROR_KEYS.groups]: s__(
      'DevopsAdoption|There was an error fetching Groups. Please refresh the page.',
    ),
    [DEVOPS_ADOPTION_ERROR_KEYS.segments]: s__(
      'DevopsAdoption|There was an error fetching Group adoption data. Please refresh the page.',
    ),
    [DEVOPS_ADOPTION_ERROR_KEYS.addSegment]: s__(
      'DevopsAdoption|There was an error enabling the current group. Please refresh the page.',
    ),
    tableHeader: {
      button: s__('DevopsAdoption|Add/remove groups'),
    },
  },
  emptyState: {
    title: s__('DevopsAdoption|Add a group to get started'),
    description: s__(
      'DevopsAdoption|DevOps adoption tracks the use of key features across your favorite groups. Add a group to the table to begin.',
    ),
    button: s__('DevopsAdoption|Add Group'),
  },
  modal: {
    addingTitle: s__('DevopsAdoption|Add/remove groups'),
    addingButton: s__('DevopsAdoption|Save changes'),
    cancel: __('Cancel'),
    namePlaceholder: s__('DevopsAdoption|My group'),
    filterPlaceholder: s__('DevopsAdoption|Filter by name'),
    error: s__('DevopsAdoption|An error occurred while saving changes. Please try again.'),
    noResults: s__('DevopsAdoption|No filter results.'),
  },
  table: {
    removeButton: s__('DevopsAdoption|Remove Group from the table.'),
  },
  deleteModal: {
    title: s__('DevopsAdoption|Confirm remove Group'),
    confirmationMessage: s__(
      'DevopsAdoption|Are you sure that you would like to remove %{name} from the table?',
    ),
    cancel: __('Cancel'),
    confirm: s__('DevopsAdoption|Remove Group'),
    error: s__('DevopsAdoption|An error occurred while removing the group. Please try again.'),
  },
  tableCell: {
    trueText: s__('DevopsAdoption|Adopted'),
    falseText: s__('DevopsAdoption|Not adopted'),
  },
};

export const DEVOPS_ADOPTION_TABLE_TEST_IDS = {
  TABLE_HEADERS: 'header',
  SEGMENT: 'segmentCol',
  ACTIONS: 'actionsCol',
  LOCAL_STORAGE_SORT_BY: 'localStorageSortBy',
  LOCAL_STORAGE_SORT_DESC: 'localStorageSortDesc',
};

export const DEVOPS_ADOPTION_SEGMENTS_TABLE_SORT_BY_STORAGE_KEY =
  'devops_adoption_segments_table_sort_by';

export const DEVOPS_ADOPTION_SEGMENTS_TABLE_SORT_DESC_STORAGE_KEY =
  'devops_adoption_segments_table_sort_desc';

export const DEVOPS_ADOPTION_GROUP_COL_LABEL = __('Group');

export const DEVOPS_ADOPTION_TABLE_CONFIGURATION = [
  {
    title: s__('DevopsAdoption|Dev'),
    tab: 'dev',
    cols: [
      {
        key: 'issueOpened',
        label: s__('DevopsAdoption|Issues'),
        tooltip: s__('DevopsAdoption|At least 1 issue opened'),
        testId: 'issuesCol',
      },
      {
        key: 'mergeRequestOpened',
        label: s__('DevopsAdoption|MRs'),
        tooltip: s__('DevopsAdoption|At least 1 MR opened'),
        testId: 'mrsCol',
      },
      {
        key: 'mergeRequestApproved',
        label: s__('DevopsAdoption|Approvals'),
        tooltip: s__('DevopsAdoption|At least 1 approval on an MR'),
        testId: 'approvalsCol',
      },
    ],
  },
  {
    title: s__('DevopsAdoption|Sec'),
    tab: 'sec',
    cols: [
      {
        key: 'securityScanSucceeded',
        label: s__('DevopsAdoption|Scanning'),
        tooltip: s__('DevopsAdoption|At least 1 security scan of any type run in pipeline'),
        testId: 'scanningCol',
      },
    ],
  },
  {
    title: s__('DevopsAdoption|Ops'),
    tab: 'ops',
    cols: [
      {
        key: 'runnerConfigured',
        label: s__('DevopsAdoption|Runners'),
        tooltip: s__('DevopsAdoption|Runner configured for project/group'),
        testId: 'runnersCol',
      },
      {
        key: 'pipelineSucceeded',
        label: s__('DevopsAdoption|Pipelines'),
        tooltip: s__('DevopsAdoption|At least 1 pipeline successfully run'),
        testId: 'pipelinesCol',
      },
      {
        key: 'deploySucceeded',
        label: s__('DevopsAdoption|Deploys'),
        tooltip: s__('DevopsAdoption|At least 1 deploy'),
        testId: 'deploysCol',
      },
    ],
  },
];

export const TRACK_ADOPTION_TAB_CLICK_EVENT = 'i_analytics_dev_ops_adoption';

export const TRACK_DEVOPS_SCORE_TAB_CLICK_EVENT = 'i_analytics_dev_ops_score';
