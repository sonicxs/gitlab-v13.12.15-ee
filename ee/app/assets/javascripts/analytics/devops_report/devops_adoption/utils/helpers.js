import { isToday } from '~/lib/utils/datetime_utility';

/**
 * A helper function which accepts the segments,
 *
 * @param {Object} params the segment data, timestamp and check for open modals
 *
 * @return {Boolean} a boolean to determine if table data should be polled
 */
export const shouldPollTableData = ({ segments, timestamp, openModal }) => {
  if (openModal) {
    return false;
  } else if (!segments.length) {
    return true;
  }

  const anyPendingSegments = segments.some((node) => node.latestSnapshot === null);
  const dataNotRefreshedToday = !isToday(new Date(timestamp));

  return anyPendingSegments || dataNotRefreshedToday;
};
