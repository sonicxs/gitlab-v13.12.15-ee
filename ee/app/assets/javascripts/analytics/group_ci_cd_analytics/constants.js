export const STAT_ERROR_PLACEHOLDER = '-';

export const TABS = ['release-statistics', 'deployment-frequency'];
