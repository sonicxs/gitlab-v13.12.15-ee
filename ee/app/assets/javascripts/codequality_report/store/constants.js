export const VIEW_EVENT_FEATURE_FLAG = 'usageDataITestingFullCodeQualityReportTotal';
export const VIEW_EVENT_NAME = 'i_testing_full_code_quality_report_total';

export const SEVERITY_SORT_ORDER = {
  unknown: 0,
  blocker: 1,
  critical: 2,
  major: 3,
  minor: 4,
  info: 5,
};
