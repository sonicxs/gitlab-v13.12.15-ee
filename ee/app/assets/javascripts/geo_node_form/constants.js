export const SELECTIVE_SYNC_SHARDS = 'selectiveSyncShards';
export const SELECTIVE_SYNC_NAMESPACES = 'selectiveSyncNamespaceIds';
export const VALIDATION_FIELD_KEYS = {
  NAME: 'name',
  URL: 'url',
  REPOS_MAX_CAPACITY: 'reposMaxCapacity',
  FILES_MAX_CAPACITY: 'filesMaxCapacity',
  CONTAINER_REPOSITORIES_MAX_CAPACITY: 'containerRepositoriesMaxCapacity',
  VERIFICATION_MAX_CAPACITY: 'verificationMaxCapacity',
  MINIMUM_REVERIFICATION_INTERVAL: 'minimumReverificationInterval',
};

export const NODE_NAME_MORE_INFO =
  'https://docs.gitlab.com/ee/administration/geo/replication/configuration.html#step-3-add-the-secondary-node';

export const NODE_INTERNAL_URL_MORE_INFO =
  'https://docs.gitlab.com/ee/user/admin_area/geo_nodes.html#using-a-different-url-for-synchronization';

export const SELECTIVE_SYNC_MORE_INFO =
  'https://docs.gitlab.com/ee/administration/geo/replication/configuration.html#selective-synchronization';

export const OBJECT_STORAGE_MORE_INFO =
  'https://docs.gitlab.com/ee/administration/geo/replication/object_storage.html';

export const REVERIFICATION_MORE_INFO =
  'https://docs.gitlab.com/ee/administration/geo/disaster_recovery/background_verification.html#repository-re-verification';

export const BACKFILL_MORE_INFO =
  'https://docs.gitlab.com/ee/user/admin_area/geo_nodes.html#geo-backfill';
