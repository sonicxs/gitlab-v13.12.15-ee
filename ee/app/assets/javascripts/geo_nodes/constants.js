export const NODE_ACTIONS = {
  TOGGLE: 'toggle',
  REMOVE: 'remove',
};

export const VALUE_TYPE = {
  PLAIN: 'plain',
  GRAPH: 'graph',
  CUSTOM: 'custom',
};

export const CUSTOM_TYPE = {
  SYNC: 'sync',
  EVENT: 'event',
  STATUS: 'status',
};

export const HEALTH_STATUS_ICON = {
  healthy: 'status_success',
  unhealthy: 'status_failed',
  disabled: 'status_canceled',
  unknown: 'status_notfound',
  offline: 'status_canceled',
};

export const HEALTH_STATUS_CLASS = {
  healthy: 'text-success-600 bg-success-100',
  unhealthy: 'text-danger-600 bg-danger-100',
  disabled: 'text-secondary-800 bg-secondary-100',
  unknown: 'text-secondary-800 bg-secondary-100',
  offline: 'text-secondary-800 bg-secondary-100',
};

export const REPLICATION_STATUS_CLASS = {
  enabled: 'gl-text-green-600 gl-bg-green-100',
  disabled: 'gl-text-orange-600 gl-bg-orange-100',
};

export const REPLICATION_STATUS_ICON = {
  enabled: 'play',
  disabled: 'pause',
};

export const TIME_DIFF = {
  FIVE_MINS: 300,
  HOUR: 3600,
};

export const STATUS_DELAY_THRESHOLD_MS = 600000;

export const HELP_INFO_URL =
  'https://docs.gitlab.com/ee/administration/geo/disaster_recovery/background_verification.html#repository-verification';

export const REPLICATION_HELP_URL =
  'https://docs.gitlab.com/ee/administration/geo/replication/datatypes.html#limitations-on-replicationverification';

export const REPLICATION_PAUSE_URL =
  'https://docs.gitlab.com/ee/administration/geo/replication/#pausing-and-resuming-replication';

export const HELP_NODE_HEALTH_URL =
  'https://docs.gitlab.com/ee/administration/geo/replication/troubleshooting.html#check-the-health-of-the-secondary-node';

export const GEO_TROUBLESHOOTING_URL =
  'https://docs.gitlab.com/ee/administration/geo/replication/troubleshooting.html';
