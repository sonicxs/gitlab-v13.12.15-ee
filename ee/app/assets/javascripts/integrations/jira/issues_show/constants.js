import { __ } from '~/locale';

export const issueStates = {
  OPENED: 'opened',
  CLOSED: 'closed',
};

export const issueStateLabels = {
  [issueStates.OPENED]: __('Open'),
  [issueStates.CLOSED]: __('Closed'),
};

export const labelsFilterParam = 'labels';
