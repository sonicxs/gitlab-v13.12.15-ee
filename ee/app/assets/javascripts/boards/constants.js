/* eslint-disable import/export */
import { issuableTypes } from '~/boards/constants';
import destroyBoardListMutation from '~/boards/graphql/board_list_destroy.mutation.graphql';
import updateBoardListMutation from '~/boards/graphql/board_list_update.mutation.graphql';

import { s__ } from '~/locale';

import boardListsQuery from './graphql/board_lists.query.graphql';
import destroyEpicBoardListMutation from './graphql/epic_board_list_destroy.mutation.graphql';
import updateEpicBoardListMutation from './graphql/epic_board_list_update.mutation.graphql';
import epicBoardListsQuery from './graphql/epic_board_lists.query.graphql';

export * from '~/boards/constants';

export const DRAGGABLE_TAG = 'div';

export const EPIC_LANE_BASE_HEIGHT = 40;

/* eslint-disable @gitlab/require-i18n-strings */
export const EpicFilterType = {
  any: 'Any',
  none: 'None',
};

export const SupportedFiltersEE = ['epicId', 'iterationTitle', 'weight'];

export const IterationFilterType = {
  any: 'Any',
  none: 'None',
  current: 'Current',
};

export const IterationIDs = {
  NONE: 0,
  CURRENT: -4,
};

export const MilestoneFilterType = {
  any: 'Any',
  none: 'None',
};

export const MilestoneIDs = {
  NONE: 0,
};

export const WeightFilterType = {
  none: 'None',
};

export const WeightIDs = {
  NONE: -2,
  ANY: -1,
};

export const GroupByParamType = {
  epic: 'epic',
};

export const ErrorMessages = {
  fetchIssueError: s__(
    'Boards|An error occurred while fetching the board issues. Please reload the page.',
  ),
  fetchEpicsError: s__(
    'Boards|An error occurred while fetching the board epics. Please reload the page.',
  ),
};

export const listsQuery = {
  [issuableTypes.issue]: {
    query: boardListsQuery,
  },
  [issuableTypes.epic]: {
    query: epicBoardListsQuery,
  },
};

export const updateListQueries = {
  [issuableTypes.issue]: {
    mutation: updateBoardListMutation,
  },
  [issuableTypes.epic]: {
    mutation: updateEpicBoardListMutation,
  },
};

export const deleteListQueries = {
  [issuableTypes.issue]: {
    mutation: destroyBoardListMutation,
  },
  [issuableTypes.epic]: {
    mutation: destroyEpicBoardListMutation,
  },
};

export default {
  DRAGGABLE_TAG,
  EpicFilterType,
};
