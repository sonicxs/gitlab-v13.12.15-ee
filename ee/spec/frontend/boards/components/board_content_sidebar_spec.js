import { GlDrawer } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import { stubComponent } from 'helpers/stub_component';
import BoardContentSidebar from '~/boards/components/board_content_sidebar.vue';
import { ISSUABLE, issuableTypes } from '~/boards/constants';
import { mockIssue, mockIssueGroupPath, mockIssueProjectPath } from '../mock_data';

describe('ee/BoardContentSidebar', () => {
  let wrapper;
  let store;

  const createStore = ({ mockGetters = {}, mockActions = {} } = {}) => {
    store = new Vuex.Store({
      state: {
        sidebarType: ISSUABLE,
        issues: { [mockIssue.id]: { ...mockIssue, epic: null } },
        activeId: mockIssue.id,
        issuableType: issuableTypes.issue,
      },
      getters: {
        activeBoardItem: () => {
          return { ...mockIssue, epic: null };
        },
        projectPathForActiveIssue: () => mockIssueProjectPath,
        groupPathForActiveIssue: () => mockIssueGroupPath,
        isSidebarOpen: () => true,
        ...mockGetters,
      },
      actions: mockActions,
    });
  };

  const createComponent = () => {
    /*
      Dynamically imported components (in our case ee imports)
      aren't stubbed automatically in VTU v1:
      https://github.com/vuejs/vue-test-utils/issues/1279.

      This requires us to additionally mock apollo or vuex stores.
    */
    wrapper = shallowMount(BoardContentSidebar, {
      provide: {
        canUpdate: true,
        rootPath: '/',
        groupId: 1,
        epicFeatureAvailable: true,
        iterationFeatureAvailable: true,
        weightFeatureAvailable: true,
      },
      store,
      stubs: {
        GlDrawer: stubComponent(GlDrawer, {
          template: '<div><slot name="header"></slot><slot></slot></div>',
        }),
      },
      mocks: {
        $apollo: {
          queries: {
            participants: {
              loading: false,
            },
            currentIteration: {
              loading: false,
            },
            iterations: {
              loading: false,
            },
          },
        },
      },
    });
  };

  beforeEach(() => {
    createStore();
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
