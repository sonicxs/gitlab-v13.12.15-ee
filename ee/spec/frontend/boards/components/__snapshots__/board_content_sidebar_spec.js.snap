// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ee/BoardContentSidebar matches the snapshot 1`] = `
<div>
  Issue details
  <board-sidebar-title-stub />
   
  <sidebar-assignees-widget-stub
    fullpath="gitlab-org/gitlab-test"
    iid="27"
    initialassignees="[object Object],[object Object]"
    issuabletype="issue"
  />
   
  <board-editable-item-stub
    class="epic"
    data-testid="sidebar-epic"
    handleoffclick="true"
    title="Epic"
  >
     
    <epics-select-stub
      canedit="true"
      class="gl-w-full"
      epicissueid="0"
      groupid="1"
      initialepic="[object Object]"
      issueid="0"
      variant="standalone"
    />
  </board-editable-item-stub>
   
  <div>
    <board-sidebar-milestone-select-stub />
     
    <div
      class="gl-mt-5"
      data-qa-selector="iteration_container"
    >
      <sidebar-editable-item-stub
        canedit="true"
        data-testid="iteration-edit-link"
        title="Iteration"
        tracking="[object Object]"
      />
    </div>
  </div>
   
  <board-sidebar-time-tracker-stub
    class="swimlanes-sidebar-time-tracker"
  />
   
  <board-sidebar-due-date-stub />
   
  <board-sidebar-labels-select-stub
    class="labels"
  />
   
  <board-editable-item-stub
    class="weight"
    data-testid="sidebar-weight"
    handleoffclick="true"
    title="Weight"
  >
     
    <gl-form-stub>
      <gl-form-input-stub
        min="0"
        placeholder="Enter a number"
        type="number"
      />
    </gl-form-stub>
  </board-editable-item-stub>
   
  <sidebar-confidentiality-widget-stub
    fullpath="gitlab-org/gitlab-test"
    iid="27"
    issuabletype="issue"
  />
   
  <sidebar-subscriptions-widget-stub
    data-testid="sidebar-notifications"
    fullpath="gitlab-org/gitlab-test"
    iid="27"
    issuabletype="issue"
  />
</div>
`;
